modAionImpl/src/org/aion/zero/impl/AionBlockchainImpl.java

```diff
@@ -664,6 +665,15 @@ public synchronized void compactState() {
             }
         }
 
+        if (ret == IMPORTED_BEST) {
+            if (TX_LOG.isDebugEnabled()) {
+                for (AionTxReceipt receipt : summary.getReceipts()) {
+                    byte[] transactionHash = receipt.getTransaction().getTransactionHash();
+                    TX_LOG.debug("Transaction: " + Hex.toHexString(transactionHash) + " was sealed into block #" + block.getNumber());
```

Comment:
do we need to do null checks on summary, receipt, and receipt.getTransaction() ?

don't want to accidentally throw an NPE while trying to construct the log message.

Reply:
> Could you add the info from line 1045 to that log as well?
> Also please update the classes to use the [correct formatting](https://github.com/aionnetwork/aion/wiki/Aion-Code-Conventions).

Thanks for the comment @AlexandraRoatis, I am assuming you mean line 1045 in AionBlockchainImpl? Im wondering why do we need this, cuz I dont think the conditions match?
https://github.com/aionnetwork/aion/blob/0ed32f107c71e255561dfadffe1f794426bcbf07/modAionImpl/src/org/aion/zero/impl/AionBlockchainImpl.java#L1045

Reply:
Thanks for the comment @aion-kelvin, for summary the NPE is already checked earlier in the method. Correct me if im wrong, as for tx and receipt I dont believe they would be throwing NPE at this stage, and if they do throw NPE, we should probably catch it.

Reply:
You meant 'returns null' when you wrote 'throwing NPE' right?

Line 634 sets summary to null and then, as far as I can see, proceeds to your newly-added code without setting it to anything else (note that after Line 634, the only thing that uses summary is after a check on ret; whereas your code will try to use summary regardless of what ret was).

As for receipt and receipt.getTransaction -- I am not totally sure what they do.  I think you're right that they're not supposed to be returning null at this stage.  But bugs happen (especially since in our code we have a habit of returning null when things don't work or things can't be found) -- it would be nice to be defensive against that.

Reply:
Okay, I've added the null checkers, I agree its always nice to do defensive programming.